tests/basic/sharedv4_svc_test.go

```diff
@@ -1307,6 +1309,158 @@ var _ = Describe("{Sharedv4SvcFunctional}", func() {
+							Env     []struct {
+								Name  string
+								Value string
+							}
+							VolumeMounts []struct {
+								Name      string
+								MountPath string `yaml:"mountPath"`
+							} `yaml:"volumeMounts"`
+						}
+						Volumes []struct {
+							Name                  string
+							PersistentVolumeClaim struct {
+								ClaimName string `yaml:"claimName"`
+							} `yaml:"persistentVolumeClaim"`
+						}
+						ImagePullSecrets []struct {
+							Name string
+						} `yaml:"imagePullSecrets"`
+					}
+				}
+			}
+		}
+		// Editing the original yaml file and bringing in new file with new name and OUTPUT_PARAMS
+		currentDir, err := os.Getwd()
+		log.FailOnError(err, "Failed to find current directory")
+		filePath := filepath.Join(currentDir, "..", "drivers", "scheduler", "k8s", "specs", "ml-workload-rwx", "query-ml-workload.yaml")
+		flag := false
+		for i := 1; i <= totalMlDeps; i++ {
+			data, err := os.ReadFile(filePath)
+			log.FailOnError(err, fmt.Sprintf("Error Reading Yaml File: %v", filePath))
+			var deployment Deployment
+			err = yaml.Unmarshal(data, &deployment)
+			log.FailOnError(err, fmt.Sprintf("Error Unmarshaling Yaml File: %v", filePath))
+			appName := fmt.Sprintf("querying-app-%d", i)
+			deployment.Metadata.Name = appName
+			deployment.Spec.Selector.MatchLabels = map[string]string{"app": appName}
+			deployment.Spec.Template.Metadata.Labels = map[string]string{"app": appName}
+			for j := range deployment.Spec.Template.Spec.Containers {
+				for k := range deployment.Spec.Template.Spec.Containers[j].Env {
+					if deployment.Spec.Template.Spec.Containers[j].Env[k].Name == "OUTPUT_FILE" {
+						deployment.Spec.Template.Spec.Containers[j].Env[k].Value = fmt.Sprintf("query_output_%d.csv", i)
+					}
+				}
+			}
+			opFilePath := filepath.Join(currentDir, "..", "drivers", "scheduler", "k8s", "specs", "ml-workload-rwx", "query-ml-workload.yaml")
+			modifiedData, err := yaml.Marshal(&deployment)
+			log.FailOnError(err, fmt.Sprintf("Error Marshaling back to Yaml File: %v", filePath))
+			err = os.WriteFile(opFilePath, modifiedData, 0644)
+			log.FailOnError(err, fmt.Sprintf("Error Writing to Yaml File: %v", filePath))
+			provider := Inst().V.String()
+			err = Inst().S.RescanSpecs(Inst().SpecDir, provider)
+			log.FailOnError(err, "Failed to rescan specs from %s for storage provider %s", Inst().SpecDir, provider)
+			// Running Several Workload pods that read the model, make some predictions and write them to shared vol
+			Inst().AppList = []string{"ml-workload-rwx"}
+			appContexts = append(appContexts, ScheduleApplicationsOnNamespace(ns, taskName)...)
+			log.Infof("Successfully created App querying-app-%d", i)
+			ValidateApplications(appContexts)
+			if i == totalMlDeps {
+				flag = true
+			}
+		}
+		if flag {
+			log.Infof("All Workload apps are now up. Will let them run for %d minutes", totalRunTime)
+			time.Sleep(time.Duration(totalRunTime) * time.Minute)
+		} else {
```

Comment:
can we add sharedv4 reallocation scenario as part of this?

Reply:
Noted. Let's do that in the next PR and a different testcase altogether for this workload.